localstack/services/dynamodb/resource_providers/aws_dynamodb_table.py

```diff
@@ -0,0 +1,414 @@
+
+    def create(
+        self,
+        request: ResourceRequest[DynamoDBTableProperties],
+    ) -> ProgressEvent[DynamoDBTableProperties]:
+        """
+        Create a new resource.
+
+        Primary identifier fields:
+          - /properties/TableName
+
+        Required properties:
+          - KeySchema
+
+        Create-only properties:
+          - /properties/TableName
+          - /properties/ImportSourceSpecification
+
+        Read-only properties:
+          - /properties/Arn
+          - /properties/StreamArn
+
+        IAM permissions required:
+          - dynamodb:CreateTable
+          - dynamodb:DescribeImport
+          - dynamodb:DescribeTable
+          - dynamodb:DescribeTimeToLive
+          - dynamodb:UpdateTimeToLive
+          - dynamodb:UpdateContributorInsights
+          - dynamodb:UpdateContinuousBackups
+          - dynamodb:DescribeContinuousBackups
+          - dynamodb:DescribeContributorInsights
+          - dynamodb:EnableKinesisStreamingDestination
+          - dynamodb:DisableKinesisStreamingDestination
+          - dynamodb:DescribeKinesisStreamingDestination
+          - dynamodb:ImportTable
+          - dynamodb:ListTagsOfResource
+          - dynamodb:TagResource
+          - dynamodb:UpdateTable
+          - kinesis:DescribeStream
+          - kinesis:PutRecords
+          - iam:CreateServiceLinkedRole
+          - kms:CreateGrant
+          - kms:Decrypt
+          - kms:Describe*
+          - kms:Encrypt
+          - kms:Get*
+          - kms:List*
+          - kms:RevokeGrant
+          - logs:CreateLogGroup
+          - logs:CreateLogStream
+          - logs:DescribeLogGroups
+          - logs:DescribeLogStreams
+          - logs:PutLogEvents
+          - logs:PutRetentionPolicy
+          - s3:GetObject
+          - s3:GetObjectMetadata
+          - s3:ListBucket
+
+        """
+        model = request.desired_state
+
+        if not request.custom_context.get(REPEATED_INVOCATION):
+            request.custom_context[REPEATED_INVOCATION] = True
+
+            if not model.get("TableName"):
+                model["TableName"] = util.generate_default_name(
+                    request.stack_name, request.logical_resource_id
+                )
+
+            if model.get("ProvisionedThroughput"):
+                model["ProvisionedThroughput"] = self.get_ddb_provisioned_throughput(model)
+
+            if model.get("GlobalSecondaryIndexes"):
+                model["GlobalSecondaryIndexes"] = self.get_ddb_global_sec_indexes(model)
+
+            properties = [
+                "TableName",
+                "AttributeDefinitions",
+                "KeySchema",
+                "BillingMode",
+                "ProvisionedThroughput",
+                "LocalSecondaryIndexes",
+                "GlobalSecondaryIndexes",
+                "Tags",
+            ]
+            create_params = util.select_attributes(model, properties)
```

Comment:
non-blocking: why use this util function rather than building the dictionary yourself? It would be nice to break away from these utils at some point, since they have a weird API and are trying to be too general

Reply:
It's probably still easier to keep the actual code more consistent using a few of these utils. In general they are only a subset now of the old ones an without their weird signatures.